Modules/_io/_iomodule.c

```diff
@@ -504,6 +506,43 @@ _io_open_impl(PyObject *module, PyObject *file, const char *mode,
     return NULL;
 }
 
+
+/*[clinic input]
+_io.text_encoding
+    encoding: object
+    stacklevel: int = 2
+    /
+
+Helper function to choose the text encoding.
+
+When encoding is not None, just return it.
+Otherwise, return the default text encoding (i.e. "locale").
+
+This function emits EncodingWarning if *encoding* is None and
+sys.flags.warn_default_encoding is true.
+
+This function can be used in APIs having encoding=None option.
+But please consider encoding="utf-8" for new APIs.
+[clinic start generated code]*/
+
+static PyObject *
+_io_text_encoding_impl(PyObject *module, PyObject *encoding, int stacklevel)
+/*[clinic end generated code: output=91b2cfea6934cc0c input=2f8639413d4a6e1a]*/
+{
+    if (encoding == NULL || encoding == Py_None) {
+        PyInterpreterState *interp = _PyInterpreterState_GET();
+        if (_PyInterpreterState_GetConfig(interp)->warn_default_encoding) {
+            PyErr_WarnEx(PyExc_EncodingWarning,
+                         "'encoding' argument is not specified", stacklevel);
```

Comment:
```suggestion
                         "'encoding' argument not specified", stacklevel);
```

Looks like this might be generated code, but if not, you'll presumably want to adapt any UI text changes made previously here; I made the minimal change above.